test.py

```diff
@@ -0,0 +1,157 @@
+        mk = self.memcache_key()
+        persistentcache.delete(mk)
+        self.delete()
+        
+class CacheTestRequestHandler(webapp2.RequestHandler):
+    def report_failure(self, msg=None):
+        if msg is not None:
+            self.response.write("Error: %s\n" % msg)
+        self.response.write("Failure")
+
+    def report_exception(self, e):
+        if e is not None:
+            import traceback
+            msg = "Exception: (%s) %s\n%s" % (type(e).__name__, e.message, traceback.format_exc())
+            self.response.write(msg)
+            logging.exception(msg)
+        self.response.write("Failure")
+    
+    def report_success(self):
+        self.response.write("Success")
+
+    def get(self):
+        self.report_failure("Abstract method CacheTestRequestHandler.get called") 
+
+class SetupTest(CacheTestRequestHandler):
+    def get(self):
+        try:
+            e = TestModel(key_name=TEST1_KEY, payload=TEST1_KEY)
+            e.putcache()
+            #e = TestModel(key_name=TEST2_KEY, payload=TEST2_KEY, blobload=TEST2_PAYLOAD)
+            #e.putcache()
+            self.report_success()
+        except Exception, e:
+            self.report_exception(e)
+
+class CleanupTest(CacheTestRequestHandler):
+    def get(self):
+        try:
+            q = TestModel.all()
+            n = q.count()
+            while n > 0:
+                for m in q.run(limit=None):
+                    m.deletecache()
+                q = TestModel.all()
+                n = q.count()
+            self.report_success()
+        except Exception, e:
+            self.report_exception(e)
+
+class TestCacheRetrieve(CacheTestRequestHandler):
+    def get(self):
+        try:
+            # Retrieve a small entity that will fit in one memcache chunk.
+            m = TestModel.getcache(TEST1_KEY)
+            assert(m is not None)
+            assert(m.payload == TEST1_KEY)
+
+            # Retrieve a large entity that will not fit into one memcache chunk.
+            #m = TestModel.getcache(TEST2_KEY)
+            #assert(m is not None)
+            #assert(m.payload == TEST2_PAYLOAD)
+
+            # Make sure that the large entity was properly chunked.
+            #test2_mc_key = "%s-overflow-0" % persistentcache.make_pckey(TEST2_KEY)
+            #assert(persistentcache.get(test2_mc_key, None, False) is not None)
+            
+            self.report_success()
+        except Exception, e:
+            self.report_exception(e)
+
+class SetupTTLTest(CacheTestRequestHandler):
+    def get(self):
+        try:
+            ttl_sec = int(self.request.get("ttl_sec", 5))
+            logging.info("Re-caching with TTL=%d seconds." % ttl_sec)
+            test1_dict = {'foo': 1, 'bar': 2}
+            test2_dict = {'bar': 1, 'foo': 2}
+            persistentcache.set("test1_dict", test1_dict, ttl_sec, True)
+            persistentcache.set("test2_dict", test2_dict, ttl_sec, False)
+            # Deliberately expire from memcache
```

Comment:
for a proper TTL test, shouldn't you wait n>5 secs instead of explicitly deleting?